lib/semver.go

```diff
@@ -12,7 +13,7 @@ func GetSemver(tfconstraint *string, mirrorURL *string) (string, error) {
 
 	listAll := true
 	tflist, _ := GetTFList(*mirrorURL, listAll) //get list of versions
-	fmt.Printf("Reading required version from constraint: %s\n", *tfconstraint)
+	fmt.Fprintf(os.Stderr, "Reading required version from constraint: %s\n", *tfconstraint)
```

Comment:
I think change from use of `fmt.Printf()` to `fmt.Fprintf(os.Stderr, …` should not be applied as an ad-hoc and partially, but the whole `tfswitch` codebase should be revisited to see where `fmt.Fprintf(os.Stderr, …` is going to be more appropriate.
Instead of switching to `fmt.Fprintf(os.Stderr, …` for this particular use case, I'd suggest to have a cmdline flag created to provide verbose or terse (and even JSON) output, so that the new feature from this PR can be used programmatically w/o it printing human-friendly messages.

Reply:
The reason behind that changes for me is that without output separation the tool is _unusable_ in automation (I am not taking into account the "parse output" approach for obvious reasons).

Reply:
My point is that such a change should be applied for all code instead of just a bit of it.
As for your reasoning (it all makes sense!) this is why I suggested to add a command line flag to "turn on" or "turn off" verbosity ("verbosity" as in printing to stdout).

@warrensbox @jukie What do you think?

Reply:
> I may prepare the appropriate change if needed

That would be great if you feel like doing that. Contributions are very welcome.
Would be great if you got `zerolog`'ing implemented with pretty logging turned on by default (since the most audience for `tfswitch` are humans) with an option to turn it off (as in use default of JSON output format) for convenience in CI/CD.
Thank you.

Reply:
@yermulnik I will wait till the @warrensbox response, as I do not have a clear vision in terms of a direction the tool suppose to move :)
My interest here is to make the CI/CD ready solution which may help with the following flow:

1. We have a terraform project
2. We do _something_ to get the `terraform` version that would be able to make `init`, `plan` and `apply` with as little configuration and duplication as possible (having `required_version` and some specific version in `.terraform-version` is still treated a duplication)
3. We use that exact version to pull the specific tag of the official Docker image with `terraform` to execute the p. 2

As of now - I've come with a https://github.com/ikorchynskyi/terraform-version-inspect as a solution - the part of version constraint functionality from `tfswitch` and some stricter parsing using `json` release version "API" w/o relying on regex parsing

Reply:
> so it is likely the plan for `v0.2` :)

Sounds great. This is how OSS comes into the world =)
Pre-built binaries and Homebrew tap would also be kewl to have 😺

Reply:
Yes, will add.
I will leave this PR for further discussion with maintainers, but as of now - I've updated `terraform-version-inspect` to use releases API, and it seems that it serves our needs quite well - already integrated in CI/CD pipelines and works as expected